components/post_view/post_list/post_list_virtualized.jsx

```diff
@@ -363,9 +371,12 @@ export default class PostList extends React.PureComponent {
                 lastViewedBottom = this.props.latestPostTimeStamp;
             }
 
+            // if we hit the bottom, we haven't just landed on the unread channel
+            const renderUnreadChannelToast = (this.state.renderUnreadChannelToast && !atBottom);
             this.setState({
                 atBottom,
                 lastViewedBottom,
+                renderUnreadChannelToast,
```

Comment:
Why store this in state instead of just checking `this.state.renderUnreadChannelToast && !this.state.atBottom` in the render method?

Reply:
renderUnreadChannelToast changes during the lifetime of the component, it is used to only show the toast for when a user lands in a channel with unreads, but once we are no longer in that situation we don't want the unreadchannel to be shown again and also to stop showin in some situations.